### PR TITLE
src/script/build-integration-branch

### PR DESCRIPTION
**Pull request by liewegas**
1. put github token in ~/.github_token
2. git checkout master
3. git pull
4. build-integration-branch <label>
5. git push ci <branchanme>

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by yuriw** [comment]
@liewegas where does 'ci' get defined?

**Comment by liewegas** [comment]
updated

File: doc/dev/testing.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+=============
+
+
+build-integration-branch
+------------------------
+
+Setup
+^^^^^
+
+#. Create a github token at `<https://github.com/settings/tokens>`_
```
Comment (tchaikov) [inline doc/dev/testing.rst]: might want to note that only the access to "public_repo" need to be checked.

File: src/script/build-integration-branch [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+for issue in j:
+    if 'pull_request' not in issue:
+        continue
+    r = requests.get(issue['pull_request']['url'] + '?access_token=' + token)
+    prs.append(json.loads(r.text or r.content))
+print("--- queried %s prs" % len(prs))
+
+# name branch
+TIME_FORMAT = '%Y-%m-%d-%H%M'
+branch = label + "-" + time.strftime(TIME_FORMAT, time.localtime())
```
Comment (yuriw) [inline src/script/build-integration-branch]: @liewegas This how my wips look now `wip-yuri-testing2-2017-09-05-1600`
can we remove `1600` ?

Comment (liewegas) [reply src/script/build-integration-branch]: that's the time (-hhmm) in case you do multiple branches in one day (i regularly do)

Comment (yuriw) [reply src/script/build-integration-branch]: ok copy

**Comment by yuriw** [comment]
@liewegas it'd be cool to have `dry-run` option for this script

**Comment by liewegas** [comment]
I'm not sure it can meaningfully dry run... you won't know if merges 
conflict without doing them.

Also it is nondestructive... all it does it modify a local branch.  You 
can always clean up with reset + checkout master.

**Comment by liewegas** [comment]
i guess it could just list the branches?

**Comment by yuriw** [comment]
yes maybe, on a second thought, if it runs one time and then second time it creates a new branch, so it can be considered "dry-run" :)

**Comment by yuriw** [comment]
yes withdrawn !

**Pull request merged**